homeassistant/components/owntracks/device_tracker.py

```diff
@@ -30,7 +30,7 @@ async def async_setup_entry(
     dev_reg = dr.async_get(hass)
     dev_ids = {
         identifier[1]
-        for device in dev_reg.devices.values()
+        for device in dev_reg.devices.get_devices_for_config_entry_id(entry.entry_id)
         for identifier in device.identifiers
         if identifier[0] == OT_DOMAIN
```

Comment:
```suggestion
```
Same as for `geofancy`: https://github.com/home-assistant/core/pull/114810#discussion_r1551349258
The identifier should always match as we filter for the `entry.entry_id`.

Reply:
f4d3dd896c0b3b4eadfdb9cb525394ef379f67de